From b3cef96befdb07aff4d4c6f1e2c5f8902ede9613 Mon Sep 17 00:00:00 2001
From: Christophe Marin <christophe@krop.fr>
Date: Sun, 9 Mar 2025 10:27:34 +0100
Subject: [PATCH] Change default settings for Qt packages

- Install examples sources by default
- Enable QT_SKIP_AUTO_PLUGIN_INCLUSION and QT_SKIP_AUTO_QML_PLUGIN_INCLUSION by default
This prevents having build time dependencies on plugins needlessly
---
 cmake/QtBuildInternalsExtra.cmake.in | 1 +
 cmake/QtPlugins.cmake.in             | 2 +-
 cmake/QtPublicPluginHelpers.cmake    | 2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/cmake/QtBuildInternalsExtra.cmake.in b/cmake/QtBuildInternalsExtra.cmake.in
index 8985f8178a9..b6eb03971d7 100644
--- a/cmake/QtBuildInternalsExtra.cmake.in
+++ b/cmake/QtBuildInternalsExtra.cmake.in
@@ -83,6 +83,7 @@ set(QT_SOURCE_TREE "@QT_SOURCE_TREE@" CACHE PATH
 # Propagate decision of building tests and examples to other repositories.
 set(QT_BUILD_TESTS @QT_BUILD_TESTS@ CACHE BOOL "Build the testing tree.")
 set(QT_BUILD_EXAMPLES @QT_BUILD_EXAMPLES@ CACHE BOOL "Build Qt examples")
+set(QT_INSTALL_EXAMPLES_SOURCES @QT_INSTALL_EXAMPLES_SOURCES@ CACHE BOOL "Install Qt examples sources")
 set(QT_BUILD_BENCHMARKS @QT_BUILD_BENCHMARKS@ CACHE BOOL "Build Qt Benchmarks")
 set(QT_BUILD_MANUAL_TESTS @QT_BUILD_MANUAL_TESTS@ CACHE BOOL "Build Qt manual tests")
 set(QT_BUILD_MINIMAL_STATIC_TESTS @QT_BUILD_MINIMAL_STATIC_TESTS@ CACHE BOOL
diff --git a/cmake/QtPlugins.cmake.in b/cmake/QtPlugins.cmake.in
index e668a4cbefe..d086316b4c7 100644
--- a/cmake/QtPlugins.cmake.in
+++ b/cmake/QtPlugins.cmake.in
@@ -6,7 +6,7 @@ include_guard(DIRECTORY)
 
 # Distributions should probably change this default.
 if(NOT DEFINED QT_SKIP_AUTO_PLUGIN_INCLUSION)
-    set(QT_SKIP_AUTO_PLUGIN_INCLUSION OFF)
+    set(QT_SKIP_AUTO_PLUGIN_INCLUSION ON)
 endif()
 
 if(NOT QT_NO_CREATE_TARGETS AND NOT QT_SKIP_AUTO_PLUGIN_INCLUSION)
diff --git a/cmake/QtPublicPluginHelpers.cmake b/cmake/QtPublicPluginHelpers.cmake
index 44ceb92e246..2a2ee1c5e92 100644
--- a/cmake/QtPublicPluginHelpers.cmake
+++ b/cmake/QtPublicPluginHelpers.cmake
@@ -649,7 +649,7 @@ macro(__qt_internal_include_qml_plugin_packages)
 
     # Distributions should probably change this default.
     if(NOT DEFINED QT_SKIP_AUTO_QML_PLUGIN_INCLUSION)
-        set(QT_SKIP_AUTO_QML_PLUGIN_INCLUSION OFF)
+        set(QT_SKIP_AUTO_QML_PLUGIN_INCLUSION ON)
     endif()
 
     set(__qt_qml_plugins_config_file_list "")
-- 
2.48.1

